hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -2173,14 +2142,21 @@ def _draw(self, ir_type, kwargs, *, observe, forced, fake_forced):
             getattr(self.observer, f"draw_{ir_type}")(
                 value, kwargs=kwargs, was_forced=was_forced
             )
-            self.__example_record.record_ir_draw(
-                ir_type,
-                value,
+            size = ir_size([value])
+            if size + self.length_ir > self.max_length_ir:
+                self.mark_overrun()
+
+            node = IRNode(
+                ir_type=ir_type,
+                value=value,
                 kwargs=kwargs,
                 was_forced=was_forced,
+                index=len(self.ir_nodes),
             )
+            self.__example_record.record_ir_draw()
+            self.ir_nodes += (node,)
             self.index_ir += 1
-            self.length_ir += ir_size([value])
+            self.length_ir += size
```

Comment:
This checks for an overrun before adding to `index_ir` or `length_ir`, matching the current symmetrical behavior for the buffer. This wasn't causing any issues...yet.